libs/indibase/inditelescope.cpp

```diff
@@ -1432,6 +1456,22 @@ bool Telescope::ISNewSwitch(const char *dev, const char *name, ISState *states,
             return true;
         }
 
+        ///////////////////////////////////
+        // Simulate Pier Side
+        ///////////////////////////////////
+        if (!strcmp(name, SimulatePierSideSP.name))
+        {
+            IUUpdateSwitch(&SimulatePierSideSP, states, names, n);
+            int index = IUFindOnSwitchIndex(&SimulatePierSideSP);
+            if (index == -1)
```

Comment:
Switch state must be set follow by IDSetSwitch(...) before returning.

Reply:
I'm not so sure what IUUpdateSwitch() does. Do I need it at all, since I set the switch values in setSimulatePierSide()?

Reply:
Aaah, now I understand what you mean. In the case that index == -1, IDSetSwitch(...) is not called. All right, I'll fix it.